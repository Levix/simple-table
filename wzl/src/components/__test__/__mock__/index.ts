import type { TableColumns, TableDataSourceType } from '../../types';
import { useDataSource } from "../../../composables/useDataSource";
import { useColumns } from "../../../composables/useColumns";
import { ref } from 'vue';

const dataSource: TableDataSourceType[] = []
for (let index = 0; index < 10; index++) {
    dataSource.push({
        key: index,
        name: `Edrward ${index}`,
        age: 18 + index,
        address: `London Park no. ${index}`,
        tags: ['nice', 'developer'],
    })
}
// 表格源数据
export { dataSource };
// 表格列配置
export const columns: TableColumns[] = [
    {
		title: "Name",
		key: "name",
	},
	{
		title: "Age",
		key: "age",
		width: 100,
		align:'center',
		sortable: {
			orderBy: "DESC",
			sorter: (curr, next) => next - curr,
		},
	},
	{
		title: "Address",
		key: "address",
	},
	{
		title: "Tags",
		key: "tags",
	},
];

export const sortIconObj = {
    ascTriangle: "▲",
    descTriangle: "▼",
    normalTriangle: "■",
}

export const a_columns: TableColumns[] = [
    {
        title: "Col_1",
        key: "Col__1",
    },
    {
        title: "Col_2",
        key: "Col__2",
        sortable: {
            orderBy: "asc"
        }
    },
    {
        title: "Col_3",
        key: "Col__3",
    },
    {
        title: "Col_4",
        key: "Col__4",
    },
]
export const d_columns: TableColumns[] = [
    {
        title: "Col_1",
        key: "Col__1",
    },
    {
        title: "Col_2",
        key: "Col__2",
        sortable: {
            orderBy: "desc"
        }
    },
    {
        title: "Col_3",
        key: "Col__3",
    },
    {
        title: "Col_4",
        key: "Col__4",
    },
]

// 处理后的配置
export const { newColumns } = useColumns(columns);
// 处理后的数据
export const { sortTableData } = useDataSource(ref(dataSource), newColumns);
// body配置
export const bodyOptions = {
    sortData: sortTableData.value,
    columns: newColumns.value,
    sortOption: {
        key: 1,
        sortStatus: 'asc',
    },
    pageSize: 10,
    curPageNumber: 1,
}

// foot配置
export const footOptions = {
    sortData: sortTableData.value,
    pageSize: 4,
}

export const newFootOptions = {
    sortData: sortTableData.value,
    pageSize: 8,
}