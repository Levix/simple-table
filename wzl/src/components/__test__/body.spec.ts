import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import WuBody from '../Body';
import { bodyOptions } from './__mock__/index';

describe('WuBody', () => {
    const wrapper = mount(WuBody, {
        props: {
            bodyOptions: bodyOptions,
        }
    })

    it('打个快照 WuBody', () => {
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.exists()).toBe(true);
    })

    it('组件是否正常渲染，参数是否正确', () => {
        expect(wrapper.find('tbody').exists()).toBe(true);

        expect(wrapper.findAll('tr')).toHaveLength(bodyOptions.sortData.length);
    })
})