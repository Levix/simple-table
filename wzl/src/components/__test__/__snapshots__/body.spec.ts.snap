// Vitest Snapshot v1

exports[`WuBody > 打个快照 WuBody 1`] = `
"<tbody class=\\"wu_body\\">
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 9</td>
    <td class=\\"body_align__center\\">27</td>
    <td class=\\"body_align__left\\">London Park no. 9</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 8</td>
    <td class=\\"body_align__center\\">26</td>
    <td class=\\"body_align__left\\">London Park no. 8</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 7</td>
    <td class=\\"body_align__center\\">25</td>
    <td class=\\"body_align__left\\">London Park no. 7</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 6</td>
    <td class=\\"body_align__center\\">24</td>
    <td class=\\"body_align__left\\">London Park no. 6</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 5</td>
    <td class=\\"body_align__center\\">23</td>
    <td class=\\"body_align__left\\">London Park no. 5</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 4</td>
    <td class=\\"body_align__center\\">22</td>
    <td class=\\"body_align__left\\">London Park no. 4</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 3</td>
    <td class=\\"body_align__center\\">21</td>
    <td class=\\"body_align__left\\">London Park no. 3</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 2</td>
    <td class=\\"body_align__center\\">20</td>
    <td class=\\"body_align__left\\">London Park no. 2</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 1</td>
    <td class=\\"body_align__center\\">19</td>
    <td class=\\"body_align__left\\">London Park no. 1</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
  <tr class=\\"wu_body__tr\\">
    <td class=\\"body_align__left\\">Edrward 0</td>
    <td class=\\"body_align__center\\">18</td>
    <td class=\\"body_align__left\\">London Park no. 0</td>
    <td class=\\"body_align__left\\">nicedeveloper</td>
  </tr>
</tbody>"
`;
